### ISSUE TITLE
Entity Framework Core 8 (.NET 8) Support

### ISSUE THREAD

**Issue by SerhiyBalan** [yv989c/BlazarTech.QueryableValues#39]
Do you plan to release a version with native support of Entity Framework Core 8 ?

Thank you very much

**Comment by yv989c** [comment]
Hey @SerhiyBalan. Yes, as soon as I have a chance. In the meantime you can use v7.

Are you experiencing any issues with v7.4?

**Comment by mboud** [comment]
Hi @yv989c ! I've switched to .Net 8 / EF Core 8 and I get this kind of error with v7.4 of the package:

`Microsoft.Data.SqlClient.SqlException (0x80131904): Invalid column name 'A'. Invalid column name 'A1'. Invalid column name 'A2'. Invalid column name 'A3'. Invalid column name 'A4'. Invalid column name 'A5'. Invalid column name 'A6'. Invalid column name 'A7'. Invalid column name 'A8'. Invalid column name 'A9'. Invalid column name 'B'. Invalid column name 'B1'. Invalid column name 'B2'. Invalid column name 'B3'. Invalid column name 'B4'. Invalid column name 'B5'. Invalid column name 'B6'. Invalid column name 'B7'. Invalid column name 'B8'. Invalid column name 'B9'. Invalid column name 'C'. Invalid column name 'C1'. Invalid column name 'C2'. Invalid column name 'C3'. Invalid column name 'C4'. Invalid column name 'C5'. Invalid column name 'C6'. Invalid column name 'C7'. Invalid column name 'C8'. Invalid column name 'C9'. Invalid column name 'D'. Invalid column name 'D1'. Invalid column name 'D2'. Invalid column name 'D3'. Invalid column name 'D4'. Invalid column name 'D5'. Invalid column name 'D6'. Invalid column name 'D7'. Invalid column name 'D8'. Invalid column name 'D9'. Invalid column name 'E'. Invalid column name 'E1'. Invalid column name 'E2'. Invalid column name 'E3'. Invalid column name 'E4'. Invalid column name 'E5'. Invalid column name 'E6'. Invalid column name 'E7'. Invalid column name 'E8'. Invalid column name 'E9'. Invalid column name 'F'. Invalid column name 'F1'. Invalid column name 'F2'. Invalid column name 'F3'. Invalid column name 'F4'. Invalid column name 'F5'. Invalid column name 'F6'. Invalid column name 'F7'. Invalid column name 'F8'. Invalid column name 'F9'. Invalid column name 'G1'. Invalid column name 'G2'. Invalid column name 'G3'. Invalid column name 'G4'. Invalid column name 'G5'. Invalid column name 'G6'. Invalid column name 'G7'. Invalid column name 'G8'. Invalid column name 'G9'. Invalid column name 'H'. Invalid column name 'H1'. Invalid column name 'H2'. Invalid column name 'H3'. Invalid column name 'H4'. Invalid column name 'H5'. Invalid column name 'H6'. Invalid column name 'H7'. Invalid column name 'H8'. Invalid column name 'H9'. Invalid column name 'I'. Invalid column name 'I1'. Invalid column name 'I2'. Invalid column name 'I3'. Invalid column name 'I4'. Invalid column name 'I5'. Invalid column name 'I6'. Invalid column name 'I7'. Invalid column name 'I8'. Invalid column name 'I9'. Invalid column name 'L'. Invalid column name 'L1'. Invalid column name 'L2'. Invalid column name 'L3'. Invalid column name 'L4'. Invalid column name 'L5'. Invalid column name 'L6'. Invalid column name 'L7'. Invalid column name 'L8'. Invalid column name 'L9'. Invalid column name 'M'. Invalid column name 'M1'. Invalid column name 'M2'. Invalid column name 'M3'. Invalid column name 'M4'. Invalid column name 'M5'. Invalid column name 'M6'. Invalid column name 'M7'. Invalid column name 'M8'. Invalid column name 'M9'. Invalid column name 'S'. Invalid column name 'S1'. Invalid column name 'S2'. Invalid column name 'S3'. Invalid column name 'S4'. Invalid column name 'S5'. Invalid column name 'S6'. Invalid column name 'S7'. Invalid column name 'S8'. Invalid column name 'S9'. Invalid column name 'Y'. Invalid column name 'Y1'. Invalid column name 'Y2'. Invalid column name 'Y3'. Invalid column name 'Y4'. Invalid column name 'Y5'. Invalid column name 'Y6'. Invalid column name 'Y7'. Invalid column name 'Y8'. Invalid column name 'Y9'.`

So for me, v7.4 is not EF Core 8 compatible.

To illustrate, here's the syntax of the method that throws this error.

```
public IReadOnlyCollection<IMainAddress> FindFromAccountIds(ISet<Guid> accountIds)
{
    IQueryable<Guid> mainAddressIds = this._dbContext.Accounts
        .Join(
            this._dbContext.AsQueryableValues(accountIds),
            account => account.Id,
            accountId => accountId,
            (account, accountId) => account.MainAddressId);

    return this.Entities
        .Join(
            mainAddressIds,
            mainAddress => mainAddress.Id,
            mainAddressId => mainAddressId,
            (mainAddress, mainAddressId) => mainAddress)
        .ToArray();
}
```

**Comment by yv989c** [comment]
Thanks for reporting this, @mboud. I'll spend some time on it this weekend.

**Comment by mboud** [comment]
Hi @yv989c ! I was using v6.x + EF Core 6 before and jumped to v7.4 + EF Core 8 hence my believing it was related to EF Core 8. Thanks for sorting this out so fast. I can still wait a bit before switching to EF Core 8 so I'll wait for future updates.

Again, thanks a lot !

**Comment by yv989c** [comment]
This has been fixed in #41.
